pkg/yurthub/poolcoordinator/coordinator.go

```diff
@@ -245,20 +267,24 @@ func (coordinator *coordinator) Run() {
 				poolCacheManager, etcdStorage, cancelEtcdStorage, err = coordinator.buildPoolCacheStore()
 				if err != nil {
 					klog.Errorf("failed to create pool scoped cache store and manager, %v", err)
+					coordinator.statusInfoChan <- electorStatusInfo
 					continue
 				}
 
-				nodeLeaseProxyClient, err := coordinator.newNodeLeaseProxyClient()
-				if err != nil {
-					klog.Errorf("cloud not get cloud lease client when becoming leader yurthub, %v", err)
-					continue
-				}
 				klog.Infof("coordinator newCloudLeaseClient success.")
 				if err := coordinator.poolCacheSyncManager.EnsureStart(); err != nil {
 					klog.Errorf("failed to sync pool-scoped resource, %v", err)
+					coordinator.statusInfoChan <- electorStatusInfo
 					continue
 				}
 				klog.Infof("coordinator poolCacheSyncManager has ensure started")
+
+				nodeLeaseProxyClient, err := coordinator.newNodeLeaseProxyClient()
+				if err != nil {
+					klog.Errorf("cloud not get cloud lease client when becoming leader yurthub, %v", err)
+					coordinator.statusInfoChan <- electorStatusInfo
+					continue
+				}
```

Comment:
Why we move `newNodeLeaseProxyClient` to here?

Reply:
nodeLeaseProxyClient is used by coordinator.delegateNodeLeaseManager.EnsureStartWithHandler, after the move from the code structure looks like a smooth logic, it`s a code block.